app/src/test/kotlin/compiler/intermediate_form/FunctionDependenciesAnalyzerTest.kt

```diff
@@ -14,8 +14,108 @@ import compiler.common.reference_collections.referenceHashSetOf
 import kotlin.test.Test
 import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
 
 class FunctionDependenciesAnalyzerTest {
+    @Test fun `test create unique identifiers one function`() {
+        /*
+         czynność no_polish_signs() {}
+         */
+        val identifierFactory = UniqueIdentifierFactory()
+        val noPolishSigns = Function("no_polish_signs", listOf(), Type.Unit, listOf())
+        val noPolishSignsIdentifier = identifierFactory.build(null, noPolishSigns.name)
+
+        val program = Program(listOf(Program.Global.FunctionDefinition(noPolishSigns)))
+        val actualIdentifiers = FunctionDependenciesAnalyzer.createUniqueIdentifiers(program)
+        val expectedIdentifiers = referenceHashMapOf(noPolishSigns to noPolishSignsIdentifier)
+        assertEquals(expectedIdentifiers, actualIdentifiers)
+    }
+
+    @Test fun `test create unique identifiers no polish signs`() {
+        /*
+         czynność polskie_znaki_są_żeś_zaiście_świetneż() {}
+         */
+        val identifierFactory = UniqueIdentifierFactory()
+        var polishSigns = Function("polskie_znaki_są_żeś_zaiście_świetneż", listOf(), Type.Unit, listOf())
+        var polishSignsIdentifier = identifierFactory.build(null, polishSigns.name)
+
+        val program = Program(listOf(Program.Global.FunctionDefinition(polishSigns)))
+        val actualIdentifiers = FunctionDependenciesAnalyzer.createUniqueIdentifiers(program)
+        val expectedIdentifiers = referenceHashMapOf(polishSigns to polishSignsIdentifier)
+        assertEquals(expectedIdentifiers, actualIdentifiers)
+    }
+
+    @Test fun `test create unique identifiers nested`() {
+        /*
+         czynność some_prefix() {
+             czynność no_polish_signs() {}
+         }
+         */
+        val identifierFactory = UniqueIdentifierFactory()
+        var noPolishSigns = Function("no_polish_signs", listOf(), Type.Unit, listOf())
+        var outerFunction = Function(
+            "some_prefix",
+            listOf(), Type.Unit, listOf(Statement.FunctionDefinition(noPolishSigns))
+        )
+        val outerFunctionIdentifier = identifierFactory.build(null, outerFunction.name)
+        val noPolishSignsIdentifier = identifierFactory.build(outerFunctionIdentifier.value, noPolishSigns.name)
+
+        val program = Program(listOf(Program.Global.FunctionDefinition(outerFunction)))
+        val actualIdentifiers = FunctionDependenciesAnalyzer.createUniqueIdentifiers(program)
+        val expectedIdentifiers = referenceHashMapOf(
+            noPolishSigns to noPolishSignsIdentifier,
+            outerFunction to outerFunctionIdentifier
+        )
+        assertEquals(expectedIdentifiers, actualIdentifiers)
+    }
+
+    @Test fun `test identical function names with accuracy to polish signs do not cause identifier conflict`() {
+        /*
+         czynność some_prefix() {
+             czynność żeś() {}
+             czynność żes() {}
+             czynność zes() {}
```

Comment:
We should add potential conflict between 'ż' and 'ź' here (reason to be found in another comment).